src/bitcoin/relayer.rs

```diff
@@ -32,8 +34,13 @@ where
 const HEADER_BATCH_SIZE: usize = 250;
 const THRESHOLD: (u64, u64) = (9, 10);
 
+#[derive(Serialize, Deserialize)]
+pub struct DepositsQuery {
+    pub receiver: String,
+}
+
 pub struct Relayer {
-    btc_client: BitcoinRpcClient,
+    btc_client: Arc<Mutex<BitcoinRpcClient>>,
```

Comment:
This should be a `tokio` sync primitive since thread blocking does not interact well with async workers. Also, it should be a `RwLock` instead of a `Mutex` since the bitcoin client methods only need an immutable receiver and we would otherwise be making all the various tasks need to wait for the lock, which would be pretty brutal on performance.

Reply:
Now remembering we create multiple relayers for the different tasks so the `Mutex` vs `RwLock` performance hit isn't as bad as I had thought, but still would hurt if there would otherwise be any concurrency in Bitcoin requests.